flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -267,20 +269,20 @@ public void close() throws Exception {
 	// ------------------------------------------------------------------------
 
 	@Override
-	public HashMap<KinesisStreamShard, SequenceNumber> snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
+	public List<Tuple2<KinesisStreamShard, SequenceNumber>> snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
 		if (lastStateSnapshot == null) {
 			LOG.debug("snapshotState() requested on not yet opened source; returning null.");
-			return null;
+			return new ArrayList<>();
```

Comment:
Its good style to avoid creating empty lists. Use `Collections.emptyList()` instead.